week2/homework/src/counter.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const fs = require('fs').promises;
+const path = require('path');
+const express = require('express');
+
+const app = express();
+app.set('port', process.env.PORT || 3000);
+
+const PORT = app.get('port');
+const FILE = path.join(__dirname, process.env.FILE || 'counters.json');
+
+const loadCounters = async () => {
+  try {
+    const counters = await fs.readFile(FILE, 'utf8');
+    return JSON.parse(counters);
+  } catch (e) {
+    return [];
+  }
+};
+
+const saveCounters = async counters => {
+  try {
+    return await fs.writeFile(FILE, JSON.stringify(counters));
+  } catch (e) {
+    return [];
+  }
+};
+
+const listCounters = async () => {
+  try {
+    return await loadCounters();
+  } catch (e) {
+    throw new Error('Please, refresh the page!');
+  }
+};
+
+const addCounter = async (id, state = 0) => {
+  const counters = await loadCounters();
+  counters.push({ id, state });
+  saveCounters(counters);
+  return { id };
+};
+
+const readCounter = async id => {
+  const counters = await loadCounters();
+  const counter = counters.find(counter => counter.id === id);
+  if (!counter) {
+    throw new Error('Counter is not found');
+  }
+  return counter;
+};
+
+const updateCounter = async id => {
+  const counters = await loadCounters();
+  const counter = counters.find(counter => counter.id === id);
+  if (!counter) {
+    throw new Error('Counter is not found');
+  }
+  ++counter.state;
+  saveCounters(counters);
+  return counter;
+};
+
+app
+  .get('/counters', async (req, res) => {
+    try {
+      res.send(await listCounters());
+    } catch (e) {
+      res.status(400).send(e.message);
```

Comment:
you may send here a 500 error since it could be a server file reading failure may be?